app/models/subscription.rb

```diff
@@ -0,0 +1,52 @@
+class Subscription < ActiveRecord::Base
+  # Associations
+  has_and_belongs_to_many :users
+  has_many :posts
+
+  # Validations
+  validate :twitter_xor_instagram_id
+  # would be good to write a validation that it has either a twitter_id OR an instagram_id
+
+  private
+    # validates that there is EITHER a twitter OR instagram id
+    # can't both be blank or both have values
+    def twitter_xor_instagram_id
+      if !(twitter_id.blank? ^ instagram_id.blank?)
+        errors.add(:ids, "Subscription must have one id")
+      end
+    end
+
+  # NOTE TO SELF: This does not work as a scope BECAUSE
+  # if it finds NOTHING it returns EVERYTHING.
+  # This is by design to make scopes chainable.
+  # Thx scopes.
+
+  # Not sure if this id == @twitter_id is 100% failsafe
+  def self.find_id(id)
+    if id == @twitter_id
```

Comment:
Where does `@twitter_id` come from? I'm not sure what this method is trying to accomplish. Let's talk about this.